salt/client/mixins.py

```diff
@@ -449,7 +449,12 @@ def print_async_event(self, suffix, event):
         # more general, since this will get *really* messy as
         # people use more events that don't quite fit into this mold
         if suffix == 'ret':  # for "ret" just print out return
-            salt.output.display_output(event['return'], None, self.opts)
+            if isinstance(event['return'], dict):
+                outputter = event['return'].get('outputter', None)
```

Comment:
You can just pop. It will return the key if it exists.
There's no need for 2 calls...